testing/example_scripts/warnings/test_group_warnings_by_message_summary.py

```diff
@@ -0,0 +1,21 @@
+import warnings
+
+import pytest
+
+
+def func():
+    warnings.warn(UserWarning("foo"))
+
+
+@pytest.fixture(params=range(20), autouse=True)
+def repeat_hack(request):
+    return request.param
+
+
+@pytest.mark.parametrize("i", range(5))
+def test_foo(i):
+    func()
+
+
+def test_bar():
+    func()
```

Comment:
I find the indirection via "example_scripts" making tests harder to read (since you need to open the other file also).

Reply:
i prefer real files over intended multiline strings, -- something i'd like on top of that is adding a way for us to run example scripts without copying instead

Reply:
Why do you prefer them?  What about the issue with not having input and expected output together?
It results in things like https://github.com/pytest-dev/pytest/pull/6813 also easily.

As a compromise I could imagine having them in the documentation instead: so they actually serve as a real world use case example in the docs.

Reply:
unlike the random strings inside of tests, one can run them directly under pytest while doing exploration and use much better debugging facilities while doing so

Reply:
"random strings" are blocks of tests that you can easily copy'n'paste and save into a file to run them manually, when you need it.
But usually it is easy enough to use `--trace` / `--pdb` with existing tests also to start with.
pytest not making it pleasant with regard to having to throw in / add `-s` etc is an issue with pytest, and making tests less readable (and actually not being checked anymore (#6814)) because of it feels wrong to me.
FWIW I've started playing with `--pytester-capture`, which would help in that regard (and with "my-master" it handles this better already in general).

Reply:
Also code is read more often than written and/or being debugged to begin with.